mlir/lib/Dialect/MIOpen/Transforms/ThreadwiseToStdlib.cpp

```diff
@@ -249,15 +257,31 @@ struct BufferLoadRewritePattern : public OpRewritePattern<BufferLoadOp> {
     auto emitLoadInstruction = [&b, loc, loadedType,
                                 source](ValueRange loadCoords) -> Value {
       Value loadedValue;
-      if (auto loadedVectorType = loadedType.dyn_cast<VectorType>()) {
-        // Issue vector load.
-        // use buffer load since the source memref is on address space 0
-        SmallVector<Value, 4> srcLowerIndicesI32;
-        for (auto v : loadCoords)
-          srcLowerIndicesI32.push_back(
-              b.create<IndexCastOp>(loc, b.getIntegerType(32), v));
-        loadedValue = b.create<gpu::MubufLoadOp>(loc, loadedType, source,
-                                                 srcLowerIndicesI32);
+
+      if (loadedType.isa<VectorType>()) {
+        VectorType loadedVectorType = loadedType.cast<VectorType>();
+        Type elementType = loadedVectorType.getElementType();
+        int64_t vectorLength = loadedVectorType.getShape()[0];
+
+        auto loadWidth = vectorLength * elementType.getIntOrFloatBitWidth();
+        bool isTooNarrow = loadWidth < 32;
+        bool isTooWide = loadWidth > 4 * 32;
+
+        // Use scalar load when load width is too narrow or too wide
```

Comment:
Can we get a test for this in the fixup?

Reply:
https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/476 tracked.

`isTooNarrow` are already tracked by new tuning parameter generated code from resnet50 e2e tests. `isTooWide` may never hit because of the input/output type of this op. My take away here is to come up with some infra beyond pass that can encode 32 and 32 * 4 correctly so it doesn't become one of those majic numbers.

Reply:
I more meant adding to the unit tests for `buffer_load` and `buffer_store` to make sure calling for something too narrow works correctly.

Reply:
I think again you have finished the work in `amdgpu-to-rocdl.mlir` and there's nothing left to do. Correct me if I'm wrong.

Reply:
Yeah, this got fixed in the rewrite, near as I can tell